### PR TITLE
msg/simple: fix missing unlock when already bind

### PR DESCRIPTION
**Pull request by yuyuyu101**
Signed-off-by: Haomai Wang <haomai@xsky.com>

### CODE REVIEW

**Comment by yuyuyu101** [comment]
caused by https://github.com/ceph/ceph/pull/12901/

**Comment by Liuchang0812** [comment]
hi, @yuyuyu101 

`AsyncMessager::client_bind` also need this patch.

File: src/msg/simple/SimpleMessenger.cc [hunk 1]
```diff
@@ -312,6 +312,7 @@ int SimpleMessenger::client_bind(const entity_addr_t &bind_addr)
   lock.Lock();
   if (did_bind) {
     assert(my_inst.addr == bind_addr);
+    lock.Unlock();
```
Comment (Liuchang0812) [inline src/msg/simple/SimpleMessenger.cc]: AsyncMessager::client_bind also need this patch.

**Comment by yuyuyu101** [comment]
thanks, done

File: src/msg/async/AsyncMessenger.cc [hunk 2]
```diff
@@ -396,6 +396,7 @@ int AsyncMessenger::client_bind(const entity_addr_t &bind_addr)
   lock.Lock();
   if (did_bind) {
     assert(my_inst.addr == bind_addr);
+    lock.Unlock();
```
Comment (tchaikov) [inline src/msg/async/AsyncMessenger.cc]: nit, better off using `Mutex::Locker`. or instead change to `std::mutex`, and use `std::lock_guard` instead.

**Comment by yuyuyu101** [comment]
@tchaikov done

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2017-02-18_17:57:32-rados-wip-kefu-testing---basic-mira/

**Pull request merged**